src/scripts/components/Accordion.ts

```diff
@@ -1,155 +1,215 @@
+        this.cancelAnimation();
+
+        return this.animate(
+            { height: [startHeight, endHeight] },
             {
                 duration: Accordion.DURATION,
-
-                easing: 'cubic-bezier(0.215, 0.61, 0.355, 1)'
+                easing: Accordion.EASING
             }
         );
+    }
 
-        if (this.animation) {
-            this.animation.onfinish = () => this.onAnimationFinish(false);
-            this.animation.oncancel = () => {
-                this.isClosing = false;
-                this.$root.classList.add(Accordion.CLASS_OPEN);
-            };
-        }
+    private prepareForAnimation(): void {
+        this.style.overflow = 'hidden';
     }
 
-    start() {
-        this.$root.style.height = `${this.$root.offsetHeight}px`;
+    private cleanupAfterAnimation(): void {
+        this.animation = null;
+        this.isClosing = false;
+        this.isExpanding = false;
+        this.$details!.style.height = '';
+        this.style.overflow = '';
+    }
 
-        window.requestAnimationFrame(() => this.expand());
+    // =============================================================================
+    // Methods
+    // =============================================================================
+    private close(): void {
+        this.isClosing = true;
+        this.$details!.classList.remove(Accordion.CLASS_ACTIVE);
+        this.prepareForAnimation();
+
+        const startHeight = `${this.$details!.offsetHeight}px`;
+        const endHeight = `${this.$summary!.offsetHeight}px`;
+
+        this.animation = this.createAnimation(startHeight, endHeight);
+        this.animation.onfinish = () => this.onAnimationFinish(false);
+        this.animation.oncancel = () => this.onCloseCancel();
+
+        this.dispatchCustomEvent('accordion-closed');
     }
 
-    expand() {
+    private open(): void {
         this.isExpanding = true;
-        this.$root.classList.add(Accordion.CLASS_OPEN);
+        this.$details!.style.height = `${this.offsetHeight}px`;
+        this.$details!.setAttribute('open', '');
+        this.$details!.classList.add(Accordion.CLASS_ACTIVE);
+        this.prepareForAnimation();
 
-        if (this.$parent) this.$parent.classList.add(Accordion.CLASS_OPEN);
+        window.requestAnimationFrame(() => {
+            const startHeight = `${this.$details!.offsetHeight}px`;
+            const endHeight = `${this.$summary!.offsetHeight + this.$content!.offsetHeight}px`;
 
-        const startHeight = `${this.$root.offsetHeight}px`;
-        const endHeight = `${this.$summary.offsetHeight + this.$content.offsetHeight}px`;
+            this.animation = this.createAnimation(startHeight, endHeight);
+            this.animation.onfinish = () => this.onAnimationFinish(true);
+            this.animation.oncancel = () => this.onExpandCancel();
+        });
 
-        if (this.animation) {
-            this.animation?.cancel();
+        this.dispatchCustomEvent('accordion-opened');
```

Comment:
The details element is either open or closed.

The purpose of the accordion component is to provide animation between those two states.

As such, I would recommend `accordion-expanded` and `accordion-collapsed` as event names. This would also allow us, if ever in the future, to introduce before events that would be named `accordion-expand` and `accordion-collapse`.